File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/history/GetRideHistorUseCase.java
package com.lattis.ellipse.domain.interactor.history;

import com.lattis.ellipse.data.network.model.response.history.RideHistoryResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.RideRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 8/16/17.
 */

public class GetRideHistorUseCase extends UseCase<RideHistoryResponse>

{

    private RideRepository rideRepository;


    @Inject
    public GetRideHistorUseCase(ThreadExecutor threadExecutor,
                             PostExecutionThread postExecutionThread,
                             RideRepository rideRepository) {
        super(threadExecutor, postExecutionThread);
        this.rideRepository = rideRepository;
    }


    @Override
    protected Observable<RideHistoryResponse> buildUseCaseObservable() {
        return this.rideRepository.getRideHistory();
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/history/GetRideHistorUseCase.kt
package com.lattis.domain.usecase.history

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.RideRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.RideHistory
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetRideHistorUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val rideRepository: RideRepository
) : UseCase<RideHistory>(threadExecutor, postExecutionThread) {
    override fun buildUseCaseObservable(): Observable<RideHistory> {
        return rideRepository.getRideHistory()
    }
}